chore: Add discourse core replacement tags for bold, italic, strike, and underline

Alteras1 opened the pull request:
fixes BBCode Tag Workarounds #106

MShultz commented on bbcode-src/tags/discourse-core-replacement.js:
```diff
@@ -0,0 +1,31 @@
+import { toNode } from "../utils/common";
+
+export const bold = (node) => {
+  return toNode("span", { class: "bbcode-b" }, node.content);
```
Just double checking--Doe these classes already exist in the repo? I don't see them as part of this PR.

Alteras1 replied:
They're part of the discourse core, as part of their default bbcode.

Alteras1 commented:
Confirming it addresses issues stated in #106 
<img width="916" alt="image" src="https://github.com/RpNation/bbcode/assets/42795314/f37dc82f-9f1f-402f-a460-176d4134a949">
